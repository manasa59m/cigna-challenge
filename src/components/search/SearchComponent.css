.search-component {
  background: #f2f2f2;
  padding: 16px;
  display: flex;
  align-items: center;
  margin-bottom: 32px;
}
.search-label {
  margin-right: 16px;
  font-weight: 600;
}

.search-input {
  padding: 8px 12px;
  width: 300px;
}
