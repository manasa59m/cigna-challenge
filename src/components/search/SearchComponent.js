import React from "react";
import "./SearchComponent.css";

const SearchComponent = ({ onSearch }) => {
  return (
    <div className="search-component">
      <label htmlFor="searchId" className="search-label">
        Search
      </label>
      <input
        id="searchId"
        data-testid="searchId"
        className="search-input"
        type="search"
        placeholder="Enter here to search"
        onChange={onSearch}
      />
    </div>
  );
};

export default SearchComponent;
