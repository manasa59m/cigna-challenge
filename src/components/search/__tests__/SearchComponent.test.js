import { shallow } from "enzyme";
import toJson from "enzyme-to-json";
import SearchComponent from "../SearchComponent";

describe("SearchComponent", () => {
  it("should render without crash", () => {
    const wrapper = shallow(<SearchComponent />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it("should call onChange function on text change in search box", () => {
      const mockHandleSearchInputValue = jest.fn();
    const wrapper = shallow(<SearchComponent onSearch={mockHandleSearchInputValue}/>);
    wrapper.find("#searchId").props().onChange({ target: { value: "su" } });
    expect(mockHandleSearchInputValue).toBeCalledWith({ target: { value: "su" } });
  });
});
