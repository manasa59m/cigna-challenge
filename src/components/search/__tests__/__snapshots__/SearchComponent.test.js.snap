// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchComponent should render without crash 1`] = `
<div
  className="search-component"
>
  <label
    className="search-label"
    htmlFor="searchId"
  >
    Search
  </label>
  <input
    className="search-input"
    data-testid="searchId"
    id="searchId"
    placeholder="Enter here to search"
    type="search"
  />
</div>
`;
